HDS-1383 Add meta tag for twitter site

KristinLBradley opened the pull request:
### :pushpin: Summary

Fix image preview for website links shared in Twitter.

**Preview:** https://hds-website-git-hds-1383-twitter-preview-hashicorp.vercel.app/
**Check Twitter share image:** https://socialsharepreview.com/

<!-- 
### :hammer_and_wrench: Detailed description
If more details are appropriate, add them here. What code changed, and why?

### :camera_flash: Screenshots
Screenshots always help, especially if this PR will change what renders to the browser -->

### :link: External links

<!-- Issues, RFC, etc. -->
Jira ticket: [HDS-1383](https://hashicorp.atlassian.net/browse/HDS-1383)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.


[HDS-1383]: https://hashicorp.atlassian.net/browse/HDS-1383?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

KristinLBradley commented on website/app/templates/head.hbs:
```diff
@@ -23,6 +23,8 @@ We've made some small customizations to simplify this compared to the default te
 
 {{! BEGIN OPENGRAPH }}
 
+<meta property="og:type" content="website" />
```
```suggestion
```

Dhaulagiri commented on website/app/templates/head.hbs:
```diff
@@ -45,6 +45,7 @@ We've made some small customizations to simplify this compared to the default te
 {{! BEGIN TWITTER }}
 
 <meta name="twitter:card" content="summary" />
+<meta name="twitter:site" content="@HashiCorp" />
```
```suggestion

{{#if this.model.twitterUsername}}
  <meta name="twitter:site" content={{this.model.twitterUsername}}>
  <meta name="twitter:creator" content={{this.model.twitterUsername}}>
{{/if}}
```

For consistency in following the way `ember-meta` [prescribes things](https://github.com/shipshapecode/ember-meta#global-config) can you add this value as a property in `config/environment.js` with the other `ember-meta` config and then add back in this logic which I had previously removed.

KristinLBradley replied:
Made this update along with reordering

Dhaulagiri reviewed (commented):
@KristinLBradley does this work for you? I tried using the preview app against twitter and the card's preview image is still blank.


If this does work, I added a suggestion for making this more consistent with how ember-meta works for other content like this.

MelSumner commented on website/app/templates/head.hbs:
```diff
@@ -45,6 +45,7 @@ We've made some small customizations to simplify this compared to the default te
 {{! BEGIN TWITTER }}
```
From experience, I learned that Twitter tags need to come before OG tags.

MelSumner commented on website/app/templates/head.hbs:
```diff
@@ -45,6 +45,7 @@ We've made some small customizations to simplify this compared to the default te
 {{! BEGIN TWITTER }}
 
 <meta name="twitter:card" content="summary" />
```
```suggestion
<meta name="twitter:card" content="summary_large_image" />
```

From my experience, this works more accurately.

KristinLBradley replied:
Done

KristinLBradley commented on website/app/templates/head.hbs:
```diff
@@ -21,45 +21,53 @@ We've made some small customizations to simplify this compared to the default te
+
+{{#if this.model.twitterUsername}}
+  <meta name="twitter:site" content={{this.model.twitterUsername}} />
+  {{!-- <meta name="twitter:creator" content={{this.model.twitterUsername}} /> --}}
```
I commented out a few tags to follow @MelSumner's example exactly and got an image to appear (together with adding a full image url).

Would it be ok to just delete these commented out tags? (Or should I try adding any back in which seem important?)

Dhaulagiri replied:
If they're not being used we can delete them

KristinLBradley replied:
Thanks, I will go ahead and just delete the ones I commented out.

KristinLBradley commented on website/config/environment.js:
```diff
@@ -30,7 +30,10 @@ module.exports = function (environment) {
-      imgSrc: '/assets/logos/share-card.jpg',
+      // TESTING below!!! was: '/assets/logos/share-card.jpg'
+      imgSrc:
+        'https://assets.codepen.io/133490/mike.jpg',
```
How should I actually do the real share-card image full url?

MelSumner replied:
it has to be the full URL because it is not relative to the social media site, if that makes sense.

Dhaulagiri replied:
@KristinLBradley I have an idea on how we can solve this. Let's pair for a few minutes when you have time.

Dhaulagiri replied:
but basically my thought is we can concatenate `imgSrc` and `url` where necessary to give the proper path to the image.

KristinLBradley replied:
I've been googling around and trying to figure out how to grab the site url but still not sure how to go about it.

Dhaulagiri replied:
let's pair on it tomorrow

Dhaulagiri commented on website/app/templates/head.hbs:
```diff
@@ -21,45 +21,53 @@ We've made some small customizations to simplify this compared to the default te
-{{#if this.model.imgSrc}}
-  <meta name="twitter:image" content={{this.model.imgSrc}} />
-{{/if}}
+<meta property="og:url" content="https://hds-website-git-hds-1383-twitter-preview-hashicorp.vercel.app/" />
```
I would move the actual url to `url` in the config file. I think for our short term purposes we could hardcode it to https://hds-website.vercel.app/ but there is likely a way we could have it be dynamic as well.

KristinLBradley replied:
Pushed update to use the variable. Let me know if I didn't use the right thing.

KristinLBradley commented:
> @KristinLBradley does this work for you? I tried using the preview app against twitter and the card's preview image is still blank.
> 
> If this does work, I added a suggestion for making this more consistent with how ember-meta works for other content like this.

Dhaulagiri commented on website/app/templates/head.hbs:
```diff
@@ -16,50 +16,48 @@ We've made some small customizations to simplify this compared to the default te
 {{#if this.model.imgSrc}}
-  <meta property="og:image" content={{this.model.imgSrc}} />
-  <meta property="og:image:type" content="image/jpg" />
+  <meta name="twitter:image" content={{this.model.imgSrc}} />
```
```suggestion
  <meta name="twitter:image" content={{concat (this.model.url + '/' + this.model.imgSrc}} />
```
something like this

KristinLBradley replied:
Thanks. The imgSrc includes "/" so I did: 
`<meta name="twitter:image" content={{concat this.model.url this.model.imgSrc}} />`

In environment.js:
```
imgSrc: '/assets/logos/share-card2.jpg',
url: 'https://hds-website-git-hds-1383-twitter-preview-hashicorp.vercel.app',
```

KristinLBradley commented on website/config/environment.js:
```diff
@@ -31,6 +31,9 @@ module.exports = function (environment) {
       imgSrc: '/assets/logos/share-card.jpg',
+      twitterUsername: '@HashiCorp',
+      // TODO: Add public site URL: https://helios.hashicorp.design
+      url: 'https://hds-website-git-hds-1383-twitter-preview-hashicorp.vercel.app',
```
@Dhaulagiri  I added this url temporarily to see if it would work but I'm still not seeing the share image show up. Does the filename of the image (imgSrc: '/assets/logos/share-card.jpg',) get fingerprinted or is there some other reason I'm overlooking?

KristinLBradley replied:
Sorry I overlooked that you could see the fingerprinting in the rendered meta tags so that's not it. Was just trying to think of some possible explanation...

KristinLBradley commented:
I still haven't figured this out but would it be worthwhile to merge what I've done so far and then keep iterating after that? I at least reordered and cleaned some things up and added a hard-coded url for the public site.

alex-ju commented:
@KristinLBradley I did a squash merge (on [a separate branch](https://github.com/hashicorp/design-system/compare/hds-1383-twitter-preview-squash?expand=1)) to obtain a single commit for all this work, then I force-pushed that commit over the existing changes to overcome the merge limitations. Please check the end result – it should be the same.

didoo reviewed (approved):
@KristinLBradley I think we can merge this PR as is now, and then we can see if/how it works in production, and if necessary open a new PR to refine things
